core/src/test/scala/kafka/tools/LogCompactionTester.scala

```diff
@@ -360,14 +358,17 @@ object LogCompactionTester {
   def readString(buffer: ByteBuffer): String = {
     val bytes = new Array[Byte](buffer.remaining)
     buffer.get(bytes)
-    new String(bytes, UTF_8)
+    Utils.utf8(bytes)
   }
 
 }
 
 case class TestRecord(topic: String, key: Int, value: Long, delete: Boolean) {
   def this(pieces: Array[String]) = this(pieces(0), pieces(1).toInt, pieces(2).toLong, pieces(3) == "d")
```

Comment:
Is this used outside of the `TestRecord.apply` method? If so, I'd move it to the companion object too and give it another name (maybe `parseComponents`). Otherwise, I'd just inline it.

Reply:
Updated the code. Pls take a look.